lib/DxilValidation/DxilContainerValidation.cpp

```diff
@@ -337,9 +337,12 @@ void PSVContentVerifier::VerifySignatureElement(
 
   PSVSignatureElement PSVSE(StrTab, IndexTab, PSVSE0);
   if (SE.IsArbitrary())
-    Mismatch |= strcmp(PSVSE.GetSemanticName(), SE.GetName());
+    Mismatch |=
+        strcmp(PSVSE.GetSemanticName(), SE.GetName()) == 0 ? false : true;
   else
-    Mismatch |= PSVSE0->SemanticKind != static_cast<uint8_t>(SE.GetKind());
+    Mismatch |= PSVSE0->SemanticKind != static_cast<uint8_t>(SE.GetKind()) == 0
```

Comment:
This change was unnecessary, plus it scares me a bit how we would be dependent on a left to right order of evaluating the comparison operations without parentheses to enforce the correct order (and the final comparison is basically `(bool) == (int)`).

Also, we could use `!= 0` everywhere instead of `== 0 ? false : true`.  That would be keeping with most of the other similar code elsewhere in this function, and it's easier to read.